external/src/main/java/com/redhat/cloud/policies/engine/handlers/TriggersHandler.java

```diff
@@ -668,13 +677,23 @@ public void findGroupMembers(RoutingContext routing) {
     public void findTriggers(RoutingContext routing) {
         routing.vertx()
                 .executeBlocking(future -> {
+                    Span serverSpan = getServerSpan(routing);
                     String tenantId = ResponseUtil.checkTenant(routing);
                     try {
                         ResponseUtil.checkForUnknownQueryParams(routing.request().params(), queryParamValidationMap.get(FIND_TRIGGERS));
-                        Pager pager = ResponseUtil.extractPaging(routing.request().params());
-                        TriggersCriteria criteria = buildCriteria(routing.request().params());
-                        Page<Trigger> triggerPage = definitionsService.getTriggers(tenantId, criteria, pager);
-                        log.debugf("Triggers: %s", triggerPage);
+                      Pager pager;
+                      try (Scope ignored = tracer.buildSpan("extractPaging").asChildOf(serverSpan).startActive(true)) {
+                          pager = ResponseUtil.extractPaging(routing.request().params());
+                        }
+                        TriggersCriteria criteria;
+                      try (Scope ignored = tracer.buildSpan("buildCriteria").asChildOf(serverSpan).startActive(true)) {
```

Comment:
buildCriteria / extractPaging shouldn't be instrumented at all. They're not external calls.

Reply:
Yeah. I mostly did it to have more lines on screen :)

Reply:
Okay, so demo purposes ;). I was wondering, would it be possible to include these as annotations? That way I could add them to the metrics-generator.groovy to check if such annotation is present (and maybe in the future I might reject a PR if a endpoint is added without instrumentation for example)

Reply:
You mean the "build span" part? It may work for methods , but certainly not inside of a long and complex method.